compiler/shared/test/diff/Defunctionalize/RecursiveFunc.mls

```diff
@@ -1,28 +1,23 @@
 :NewDefs
 
-:mono
 fun fac(n) = 
   if (n > 1) then fac(n - 1) * n else 1
 fac(5)
-//│ Lifted:
-//│ TypingUnit {
-//│   fun fac$1 = (n,) => {if ('(' >(n, 1,) ')') then *(fac$1(-(n, 1,),), n,) else 1}
-//│   Code(List(fac$1(5,)))
-//│ }
-//│ Mono:
-//│ TypingUnit {
-//│   fun fac$1 = (n,) => {if ('(' >(n, 1,) ')') then *(fac$1(-(n, 1,),), n,) else 1}
-//│   fun main$$1 = () => fac$1(5,)
-//│   Code(List(main$$1()))
-//│ }
-//│ fun fac$1: Int -> Int
-//│ fun main$$1: () -> Int
+//│ 
+//│ Simpledef:
+//│ {fun fac = (n,) => {if ('(' >(n, 1,) ')') then *(fac(-(n, 1,),), n,) else 1}
+//│  fac(5,)}
+//│ End simpledef
+//│ 
+//│ fun fac: Int -> Int
 //│ Int
 //│ res
 //│     = 120
 
-// FIXME: Strange syntax
-:mono
+// TODO: Support for specialized pattern matching types
```

Comment:
Constraining this would require implementing intersections and negative types, which I'm not sure if is within this scope of this PR for now.